pkg/devices/power_meter.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package devices
+
+import "fmt"
+
+// Joules represents an energy value in SI unit of Joule.
+// The value can additionally have an SI prefix of Micro, Milli etc
+// not captured in this type.
+// ** Do not assume that the value contained is a multiple of 1 Joule **,
+// it could be MicroJoules, or MilliJoules according to the interface
+// or method which returns it.
+type Joules uint64
+
+// String prints Joules withou units
+func (j Joules) String() string {
+	return fmt.Sprintf("%d", j)
+}
+
+// FromMicroJoules converts a value from microjoules (μJ) to joules (J).
+func FromMicroJoules(μj Joules) Joules {
+	return μj / 1_000_000
```

Comment:
Might be better to instead use uj for better readability. Unicode might also confuse linters and tools.

Reply:
Yes, please :) `uJ`

Reply:
updated